Prolific/Prolific/Utils/Persistence/DAO.m

```diff
@@ -84,6 +85,22 @@ - (void)getUserWithId:(NSString *)userId
     }];
 }
 
+- (void)followProject:(Project *)project
+              forUser:(User *)user
+           completion:(void(^)(NSError *error))completion {
+    FIRCollectionReference *const projsFollowingRef = [[[_db collectionWithPath:kUsersKey] documentWithPath:user.userId] collectionWithPath:kProjectsFollowingKey];
+    
+    NSDictionary *const data = @{
+        kNameKey: project.projectId
+    };
+    
+    [[projsFollowingRef documentWithPath:project.projectId] setData:data
+                                                              merge:YES
+                                                         completion:^(NSError *error) {
+        error ? completion(error) : completion(nil);
```

Comment:
nice use of turnary. But you don't need it in this case. Your logic reads like this "If error is not nil, provide error to the completion block, otherwise provide nil"

But you can just pass in error. If it is nil it will pass nil and if not it will provide the error. In fact, you could probably just pass "completion" into `documentWithPath`